vaadin-notification-flow-parent/vaadin-notification-flow-integration-tests/src/main/java/com/vaadin/flow/component/notification/tests/NotificationPreserveOnRefreshPage.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2000-2022 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.notification.tests;
+
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.NativeButton;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.notification.Notification;
+import com.vaadin.flow.router.PreserveOnRefresh;
+import com.vaadin.flow.router.Route;
+
+@PreserveOnRefresh
+@Route("notification-preserveonrefresh")
+public class NotificationPreserveOnRefreshPage extends Div {
+
+    public NotificationPreserveOnRefreshPage() {
+        setSizeFull();
+
+        NativeButton btn = new NativeButton("Test");
+        Notification notification = new Notification("test");
+        notification.addOpenedChangeListener(event -> {
+            if (!notification.isOpened()) {
+                Span closed = new Span("closed");
+                closed.setId("closed");
```

Comment:
nit: a new span instance is added every time the notification is closed, and each instance will have the same id. Rather add the element to the UI on init and update its text on notification close